ckanext/stadtzhtheme/fanstatic/ogdzh_autocomplete.js

```diff
@@ -6,41 +6,52 @@
 // ----------------------------------------------------------------------------
 // This js module implements autocompletion for the dataset forms
 // - it uses jquery ui autocompletion
-// - it uses getJSON to get the data from the api
-// - it relies on the api action ogdzh_autosuggest
+// - it uses getJSON to get the data from the ckan api
+// - it relies on the action ogdzh_autosuggest
 // - it serves 2 forms that are on the same page:
-//   therefore for 2 id selectors are served
+//   therefore there are two id selectors that are served
+// - the main search form on the dataset page may have context
+//   that is extracted and added as parameter to the api call
 
 "use strict";
 
 ckan.module('ogdzh_autocomplete', function ($) {
   return {
     initialize: function () {
         var getData = function (request, response) {
-            console.log("new search");
+            // api action to use
             var url = '/api/3/action/ogdzh_autosuggest';
+            // search term
             var params = {q: request.term};
-            // check if any filters/facets are set and send them along
-            var values = [];
-            var filtered = $('.filtered');
-
-            //console.log(filtered);
-            for (var i = 0; i < filtered.length; i++) {
-                var value = filtered[i].innerText.trim();
-                if (value !== "Creative Commons CCZero") {
-                    values.push(value);
+            // determine which form is in focus
+            var searchForm = $(':focus')[0].id;
+            if (searchForm === 'ogdzh_search') {
+                // only for the main dataset form:
+                // check if any filters/facets are set and send them along
+                var values = [];
+                $("#dataset-search-form input[name='groups']").each(function (elem) {
+                    values.push($(this).val());
+                });
+                $("#dataset-search-form input[name='license_id']").each(function (elem) {
+                    values.push($(this).val());
+                });
+                $("#dataset-search-form input[name='res_format']").each(function (elem) {
+                    values.push($(this).val());
+                });
+                $("#dataset-search-form input[name='tags']").each(function (elem) {
+                    values.push($(this).val());
```

Comment:
now that I see that we have to go through each filter by hand, I think it makes sense to add a note to the README-section that new filters will have to be added here before they are passed along with the autosuggestion-search to save our future selves from having to debug this ;)